sbncode/FluxReader/DK2NuInterface.cxx

```diff
@@ -0,0 +1,291 @@
+#include "DK2NuInterface.h"
+#include "dk2nu/tree/dk2nu.h"
+#include "dk2nu/tree/dkmeta.h"
+#include "dk2nu/tree/NuChoice.h"
+#include "dk2nu/tree/calcLocationWeights.h"
+
+#include "TFile.h"
+#include "TTree.h"
+
+#include <iomanip>
+
+namespace fluxr {
+  DK2NuInterface::DK2NuInterface()
+  {
+  }
+
+  DK2NuInterface::~DK2NuInterface()
+  {
+  }
+
+  void DK2NuInterface::SetRootFile(TFile* rootFile)
+  {
+    fDk2NuTree=dynamic_cast<TTree*>(rootFile->Get("dk2nuTree"));
+    fDkMetaTree=dynamic_cast<TTree*>(rootFile->Get("dkmetaTree"));
+
+    fDk2Nu = new bsim::Dk2Nu;
+    fDkMeta= new bsim::DkMeta;
+    fNuChoice=new bsim::NuChoice;
+    fDk2NuTree->SetBranchAddress("dk2nu",&fDk2Nu);
+    fDkMetaTree->SetBranchAddress("dkmeta",&fDkMeta);
+
+    fNEntries=fDk2NuTree->GetEntries();
+    fDkMetaTree->GetEntry(0);
+    fRun=fDkMeta->job;
+    fPOT=fDkMeta->pots;
+  }
+
+  void DK2NuInterface::Init(fhicl::ParameterSet const & ps)
+  {
+    //code to handle flux window stolen from GENIE_R21210/src/FluxDrivers/GNuMIFlux.cxx
+    //rotation matrix
+    try {
+      std::vector<double> rotmat=ps.get<std::vector<double> >("rotmatrix");
+      TVector3 newX,newY,newZ;
+      if (rotmat.size()==9) {
+	std::cout<<"Matrix defined with new axis values"<<std::endl;
+	newX=TVector3(rotmat[0], rotmat[1], rotmat[2]);
+	newY=TVector3(rotmat[3], rotmat[4], rotmat[5]);
+	newZ=TVector3(rotmat[6], rotmat[7], rotmat[8]);
+      } else if (rotmat.size()==6) {
+	  std::cout<<"Matrix defined with theta phi rotations"<<std::endl;
+	  newX = AnglesToAxis(rotmat[0],rotmat[1]);
+	  newY = AnglesToAxis(rotmat[2],rotmat[3]);
+	  newZ = AnglesToAxis(rotmat[4],rotmat[5]);
+	  fTempRot.RotateAxes(newX,newY,newZ);
+	  fBeamRotXML = fTempRot;  //.Inverse();
+      }
+      fTempRot.RotateAxes(newX,newY,newZ);
+      fBeamRotXML = fTempRot.Inverse();
+      fBeamRot    = TLorentzRotation(fBeamRotXML);
+      fBeamRotInv = fBeamRot.Inverse();
+    } catch (std::exception& e) {
+      std::cout<<e.what()<<std::endl;
+      std::cout<<"These are not numbers"<<std::endl;
```

Comment:
I think the try/catch can just be around the `ps.get()`. If something goes wrong any further through this, it's a different cause.
And you should definitely call `abort()` after this happens. If the matrix is not configured properly, it's no good printing a vague message to the screen and then continuing with random element values. Similar comments for various printouts further down.